src/main/kotlin/com/sainsburys/k2zpl/command/FieldData.kt

```diff
@@ -2,15 +2,33 @@ package com.sainsburys.k2zpl.command
 
 import com.sainsburys.k2zpl.builder.ZplBuilder
 
-internal data class FieldData(val data: String) : ZplCommand {
+internal data class FieldData(val data: String, val replaceNewlines: Boolean = false) : ZplCommand {
     override val command: CharSequence = "^FD"
     override val parameters: LinkedHashMap<CharSequence, Any?> = linkedMapOf("d" to data)
```

Comment:
It's just occurred to me that `parameters` should probably be of type `Map` on the `ZplCommand` interface so it becomes read-only and we can replace the implementation in the future if needed.

WDYT?

Reply:
So the reason is I wanted to enforce the use of an ordered map in the subclasses of `ZplCommand` by using `LinkedHashMap`. Do you know if there's another way to achieve this?

Reply:
Not really, short of upgrading to Java 21 and using [sequenced collection interfaces](https://openjdk.org/jeps/431), if Android even supports them yet.

But the docs for `mapOf` and `mutableMapOf` say that they preserve entry order, and given they're used so often, I'd say this is a worthwhile tradeoff.

Also I'm a bit biased as I value immutability heavily over map iteration order and I'm used to maps having a non-deterministic iteration order in Golang. 😉 

WDYT?

Reply:
I suppose another way would be to introduce a new method `addParameters` that accepts a varargs of pairs and then adds them into a `LinkedHashMap` behind the scenes, with `override val parameters` becoming a non-override getter that is a read-only view into this internal `LinkedHashMap`, a la Java's `Collections::unmodifiableMap`.

Reply:
Yeah, actually the second option (`addParameters` and making `parameters` a read-only view) sounds like the way to go! ✨

Reply:
@jbduncan finally got around to this, let me know what you think 😄

Reply:
@itsmattking Sorry, I realise I wasn't clear! I've just made PR https://github.com/sainsburys-tech/k2zpl/pull/20 which shows what I have in mind.